pages/index.js

```diff
@@ -39,18 +57,60 @@ export default function HomePage() {
     setSuccessMessage("Transaction successfully deleted!");
   }
 
+  function handleEditTransaction(updatedTransaction) {
+    setTransactionsList(
+      transactionsList.map((transaction) =>
+        transaction.id === updatedTransaction.id
+          ? updatedTransaction
+          : transaction
+      )
+    );
+    setSuccessMessage("Transaction successfully updated!");
+  }
+
+  function handleOpenEditMode(transaction) {
+    setIsEditing(true);
+    setUpdatedTransaction(transaction);
+    setIsModalOpen(true);
+  }
+
+  function openModal() {
+    setIsModalOpen(true);
+  }
+
+  function closeModal() {
+    setIsModalOpen(false);
+  }
+
   return (
     <>
       <Header />
       <main>
         <StyledTitle>Transactions</StyledTitle>
-        <TransactionForm onAdd={handleAddTransaction} />
+        <EditTransactionModal
+          isModalOpen={isModalOpen}
+          onCloseModal={closeModal}
+          isEditing={isEditing}
+          transaction={updatedTransaction}
+          onHandleEditTransaction={handleEditTransaction}
+          formatNumber={formatNumber}
+        ></EditTransactionModal>
+        <TransactionForm
+          formHeader="Add a new transaction"
+          buttonText="Add"
+          onAdd={handleAddTransaction}
+        />
         {successMessage && (
           <StyleSuccessMessage>{successMessage}</StyleSuccessMessage>
         )}
         <TransactionsList
           handleDeleteTransaction={handleDeleteTransaction}
+          handleEditTransaction={handleEditTransaction}
           transactions={transactionsList}
+          handleOpenEditMode={handleOpenEditMode}
+          openModal={openModal}
+          onCloseModal={closeModal}
+          formatNumber={formatNumber}
```

Comment:
please define this function in a lib/utils file and import it directly. You don't need to use props for this.